src/Microsoft.Health.Dicom.Blob/Features/ExternalStore/ExternalStoreHealthExpiryHttpPipelinePolicy.cs

```diff
@@ -0,0 +1,59 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using EnsureThat;
+using Microsoft.Health.Dicom.Blob.Utilities;
+
+namespace Microsoft.Health.Dicom.Blob.Features.ExternalStore;
+
+internal class ExternalStoreHealthExpiryHttpPipelinePolicy : HttpPipelinePolicy
+{
+    private readonly ExternalBlobDataStoreConfiguration _externalStoreOptions;
+    private readonly Regex _healthCheckRegex;
+    private const string GuidRegex = @"[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}";
+
+    public ExternalStoreHealthExpiryHttpPipelinePolicy(ExternalBlobDataStoreConfiguration externalStoreOptions)
+    {
+        _externalStoreOptions = EnsureArg.IsNotNull(externalStoreOptions, nameof(externalStoreOptions));
+        EnsureArg.IsNotNull(externalStoreOptions.BlobContainerUri, nameof(externalStoreOptions.BlobContainerUri));
+        EnsureArg.IsNotNull(externalStoreOptions.StorageDirectory, nameof(externalStoreOptions.StorageDirectory));
+        EnsureArg.IsNotNull(externalStoreOptions.HealthCheckFilePath, nameof(externalStoreOptions.HealthCheckFilePath));
+
+        UriBuilder uriBuilder = new UriBuilder(_externalStoreOptions.BlobContainerUri);
+        uriBuilder.Path = Path.Combine(uriBuilder.Path, _externalStoreOptions.StorageDirectory, _externalStoreOptions.HealthCheckFilePath);
+
+        string healthCheckPathRegex = Regex.Escape(uriBuilder.Uri.AbsoluteUri);
+        string txtRegex = Regex.Escape(".txt");
+        _healthCheckRegex = new Regex($"^{healthCheckPathRegex}{GuidRegex}{txtRegex}$", RegexOptions.Compiled);
```

Comment:
nit: include the culture invariant option

Reply:
They don't make the constructor with `CultureInfo` public for some reason... had to change the options to `RegexOptions.InvariantCulture`

Reply:
That's what I meant